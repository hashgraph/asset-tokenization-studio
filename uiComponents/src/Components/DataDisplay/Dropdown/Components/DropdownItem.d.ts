/// <reference types="react" />
import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";
import type { MenuItemProps as ChakraMenuItemProps } from "@chakra-ui/react";
export interface DropdownItemProps extends ChakraMenuItemProps {
    label: string;
    icon?: PhosphorIconProps["as"];
    isDisabled?: boolean;
    isActive?: boolean;
}
export declare const DropdownItem: ({ label, icon, isDisabled, isActive, ...props }: DropdownItemProps) => JSX.Element;
