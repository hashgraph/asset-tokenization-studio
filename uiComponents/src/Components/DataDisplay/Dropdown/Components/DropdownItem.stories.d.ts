declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./DropdownItem").DropdownItemProps>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./DropdownItem").DropdownItemProps>;
export declare const WithoutIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./DropdownItem").DropdownItemProps>;
