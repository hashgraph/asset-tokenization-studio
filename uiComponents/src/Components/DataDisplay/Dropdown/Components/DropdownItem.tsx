import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import type { MenuItemProps as ChakraMenuItemProps } from "@chakra-ui/react";
import { MenuItem as ChakraMenuItem } from "@chakra-ui/react";
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import type { DropdownThemeStyle } from "../Dropdown";

export interface DropdownItemProps extends ChakraMenuItemProps {
  label: string;
  icon?: PhosphorIconProps["as"];
  isDisabled?: boolean;
  isActive?: boolean;
}
export const DropdownItem = ({
  label,
  icon,
  isDisabled,
  isActive,
  ...props
}: DropdownItemProps) => {
  const styles = useChakraMultiStyleConfig("Dropdown", {
    isActive,
    isDisabled,
  }) as DropdownThemeStyle;

  return (
    <ChakraMenuItem
      data-testid={"dropdown-item"}
      disabled={isDisabled}
      {...props}
      sx={styles.itemContainer}
    >
      {icon && (
        <PhosphorIcon
          data-testid="dropdown-item-icon"
          as={icon}
          size="xxs"
          weight={styles.itemIconWeight}
        />
      )}
      <Text sx={styles.itemLabel}>{label}</Text>
    </ChakraMenuItem>
  );
};
