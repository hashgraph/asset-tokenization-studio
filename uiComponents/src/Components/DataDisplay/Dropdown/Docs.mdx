import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as DropdownStories from "./Dropdown.stories";

<Meta of={DropdownStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=2092%3A22397&mode=design&t=k4jSCm0Q182mRyVn-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `Dropdown.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Dropdown: ConfigDropdown,
  },
};
```

```tsx dark
const ConfigDropdown: DropdownThemeConfiguration = {
  parts: dropdownPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"wrapper" | "container" | "itemIconWeight" | "itemLabel" | "itemContainer"` (`dropdownPartsList`)

## How to use

### Dropdown

Make sure to wrap the `Dropdown` component inside a `Menu` component from `@chakra-ui`

The trigger to open it should be wrapped in a `MenuButton` from `@chakra-ui`

```tsx dark
<Menu>
  <MenuButton>
    <Avatar />
  </MenuButton>
  <Dropdown>
    <DropdownItem label="Item" isActive />
    <DropdownItem label="Item" />
    <DropdownItem label="Item" />
    <DropdownItem label="Item" />
  </Dropdown>
</Menu>
```

### DropdownItem

Use this component inside the `Dropdown` component.

```tsx dark
<DropdownItem icon={Cookie} label="Item" />
```

### I only want the style of the dropdown

If you only want the look and feel of the component, but you want to implement it in other component because the `Dropdown` doesn't fit your needs, you can use:

```tsx dark
const styles = useChakraMultiStyleConfig("Dropdown", {
  variant,
}) as DropdownThemeStyle;
```

## Props

<ArgTypes of={DropdownStories} />

## Stories

<Stories title="" />
