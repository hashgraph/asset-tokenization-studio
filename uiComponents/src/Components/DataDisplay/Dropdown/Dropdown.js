var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { MenuList as ChakraMenuList, Stack as ChakraStack, } from "@chakra-ui/react";
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
export var dropdownPartsList = ["wrapper", "container", "itemIconWeight", "itemLabel", "itemContainer"];
export var Dropdown = function (_a) {
    var children = _a.children, variant = _a.variant, props = __rest(_a, ["children", "variant"]);
    var styles = useChakraMultiStyleConfig("Dropdown", {
        variant: variant,
        hasMaxWidth: true
    });
    return (React.createElement(ChakraMenuList, __assign({ position: "relative", sx: styles.wrapper }, props),
        React.createElement(ChakraStack, { spacing: 1, sx: styles.container }, children)));
};
