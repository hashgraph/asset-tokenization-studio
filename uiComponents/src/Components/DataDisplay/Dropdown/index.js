export * from "./Dropdown";
export * from "./Components";
