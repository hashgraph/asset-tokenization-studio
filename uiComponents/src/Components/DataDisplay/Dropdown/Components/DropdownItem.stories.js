var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { DropdownItem } from "./DropdownItem";
import { CurrencyCircleDollar } from "@phosphor-icons/react";
import { Menu } from "@chakra-ui/react";
var meta = {
    title: "Design System/Data Display/Dropdown/DropdownItem",
    component: DropdownItem,
    argTypes: {
        isActive: {
            control: "boolean",
            description: "Boolean to specify if the item is active."
        },
        isDisabled: {
            control: "boolean",
            description: "Boolean to specify if the item is disabled."
        },
        icon: { control: false, description: "Icon to show in the item" }
    },
    parameters: {
        docs: {}
    },
    args: {
        label: "Item molecule",
        icon: CurrencyCircleDollar
    }
};
export default meta;
var Template = function (args) {
    return (React.createElement(Menu, null,
        React.createElement(DropdownItem, __assign({}, args))));
};
export var Default = Template.bind({});
Default.args = {};
export var WithoutIcon = Template.bind({});
WithoutIcon.args = {
    icon: null
};
