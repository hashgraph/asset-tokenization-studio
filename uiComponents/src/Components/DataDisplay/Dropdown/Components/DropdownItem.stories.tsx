import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { DropdownItem } from "./DropdownItem";
import { CurrencyCircleDollar } from "@phosphor-icons/react";
import { Menu } from "@chakra-ui/react";

const meta = {
  title: "Design System/Data Display/Dropdown/DropdownItem",
  component: DropdownItem,
  argTypes: {
    isActive: {
      control: "boolean",
      description: "Boolean to specify if the item is active.",
    },
    isDisabled: {
      control: "boolean",
      description: "Boolean to specify if the item is disabled.",
    },
    icon: { control: false, description: "Icon to show in the item" },
  },
  parameters: {
    docs: {},
  },
  args: {
    label: "Item molecule",
    icon: CurrencyCircleDollar,
  },
} as Meta<typeof DropdownItem>;
export default meta;

const Template: StoryFn<typeof DropdownItem> = (args) => {
  return (
    <Menu>
      <DropdownItem {...args} />
    </Menu>
  );
};

export const Default = Template.bind({});
Default.args = {};

export const WithoutIcon = Template.bind({});
WithoutIcon.args = {
  icon: null,
};
