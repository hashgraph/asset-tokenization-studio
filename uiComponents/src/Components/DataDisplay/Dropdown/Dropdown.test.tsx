import { render } from "@/test-utils";
import React from "react";
import type { DropdownProps } from "./Dropdown";
import { Dropdown } from "./Dropdown";
import { Menu } from "@chakra-ui/react";
import { DropdownItem } from "./Components";

describe(`<Dropdown />`, () => {
  const factoryComponent = (props?: DropdownProps) =>
    render(
      <Menu isOpen={true}>
        <Dropdown>
          <DropdownItem label="Dropdow Item" isActive />
          <DropdownItem label="Dropdow Item" />
          <DropdownItem label="Dropdow Item" />
          <DropdownItem label="Dropdow Item" />
        </Dropdown>
      </Menu>
    );

  test("renders correctly", () => {
    const component = factoryComponent();
    expect(component.getAllByTestId("dropdown-item")).toHaveLength(4);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
