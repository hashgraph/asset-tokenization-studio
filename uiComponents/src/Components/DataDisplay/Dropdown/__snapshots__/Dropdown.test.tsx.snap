// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-r6z5ec"
    style="visibility: visible; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
  >
    <div
      aria-orientation="vertical"
      class="chakra-menu__menu-list css-12srbko"
      id="menu-list-:r1:"
      role="menu"
      style="transform-origin: var(--popper-transform-origin); visibility: visible; opacity: 1; transform: none;"
      tabindex="-1"
    >
      <div
        class="chakra-stack css-6ekxde"
      >
        <button
          class="chakra-menu__menuitem css-mbp4z0"
          data-index="0"
          data-testid="dropdown-item"
          id="menu-list-:r1:-menuitem-:r2:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="css-1a1j0go"
          >
            Dropdow Item
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-jtq5p1"
          data-index="1"
          data-testid="dropdown-item"
          id="menu-list-:r1:-menuitem-:r3:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="css-1a1j0go"
          >
            Dropdow Item
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-jtq5p1"
          data-index="2"
          data-testid="dropdown-item"
          id="menu-list-:r1:-menuitem-:r4:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="css-1a1j0go"
          >
            Dropdow Item
          </p>
        </button>
        <button
          class="chakra-menu__menuitem css-jtq5p1"
          data-index="3"
          data-testid="dropdown-item"
          id="menu-list-:r1:-menuitem-:r5:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <p
            class="css-1a1j0go"
          >
            Dropdow Item
          </p>
        </button>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
